src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/aggregators/PersistableSlidingWindow.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.aggregators;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.persistence.FileRotate;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Iterator;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.LineIterator;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * PersistableSlidingWindow is a SlidingWindow which can have its data written to and read from disk
+ */
+public class PersistableSlidingWindow extends SlidingWindow<SlidingWindowData> {
+  private static final Logger LOG = LogManager.getLogger(PersistableSlidingWindow.class);
+  private static final ObjectMapper objectMapper = new ObjectMapper();
+
+  private Path pathToFile;
+
+  public PersistableSlidingWindow(int slidingWindowSize,
+                                  TimeUnit timeUnit,
+                                  Path filePath) {
+    super(slidingWindowSize, timeUnit);
+    this.pathToFile = filePath;
+    try {
+      if (Files.exists(pathToFile)) {
+        load(pathToFile);
+      }
+    } catch (IOException e) {
+      LOG.error("Couldn't create file {} to perform young generation tuning", pathToFile, e);
+      throw new IllegalArgumentException("Couldn't create or read a file at " + filePath);
```

Comment:
Do we really need to wrap this into an `IllegalArgumentException`? We could leave this as the checked `IOException`. It will ensure that callers handle it gracefully.

Reply:
file could be accidentally deleted by operator or even some other process(i.e. log pusher). should we create a new file in case IOException occurs ?

Reply:
I've changed this not to throw. If we can't load data on construction we'll just log an error.

A caller who is expecting data to be loaded can check if the window is empty.
A deleted/moved file issue will be resolved on the next call to write() as it includes an atomic move operation.